packages/components/src/search/test/index.js

```diff
@@ -22,6 +26,96 @@ describe( 'Search', () => {
 		expect( queryAllByRole( 'option' ) ).toHaveLength( 0 );
 	} );
 
+	describe( 'with `type="custom"`', () => {
+		describe( 'renders options given in `autocompleter.options`', () => {
+			it( 'as a static array', async () => {
+				const customAutocompleter = {
+					...productsAutocompleter,
+					// Set the options as a static array.
+					options: [
+						{ name: 'Apple', id: 1 },
+						{ name: 'Orange', id: 2 },
+						{ name: 'Grapes', id: 3 },
+					],
+				};
+				const { getByRole, queryAllByRole } = render(
+					<Search
+						type="custom"
+						autocompleter={ customAutocompleter }
+					/>
+				);
+				// Emulate typing to render available options.
+				userEvent.type( getByRole( 'combobox' ), 'A' );
+				// Wait for async options procesing.
+				await delay( 1000 );
```

Comment:
Rather than use a delay, you can use `waitFor`: https://github.com/woocommerce/woocommerce-admin/blob/3362133fa0b68f15f42e3dce614c3575f7ac27e2/packages/components/src/select-control/test/index.js#L39-L46

Reply:
Thanks, TIL.